common/http/platform/beast/http.cpp

```diff
@@ -565,26 +615,37 @@ void Client::ReadHeaderHandler(const error_code &ec, size_t num_read) {
 		return;
 	}
 
-	if (response_ && !response_->body_async_reader_) {
-		// If there is no registered reader, then we need to schedule the download
-		// ourselves. Else the reader will do it.
-		ReadNextBodyPart(body_buffer_.size());
+	if (body_status_ == BodyReadingStatus::None) {
+		CallErrorHandler(MakeError(BodyIgnoredError, ""), request_, body_handler_);
 	}
 }
 
 void Client::AsyncReadNextBodyPart(
 	vector<uint8_t>::iterator start, vector<uint8_t>::iterator end, io::AsyncIoHandler handler) {
+	if (body_status_ == BodyReadingStatus::None
+		|| body_status_ == BodyReadingStatus::ReaderCreated) {
+		body_status_ = BodyReadingStatus::InProgress;
+	}
```

Comment:
When can we transition from `None` to `InProgress`? I thought that calling `MakeBodyAsyncReader` is mandatory before starting the read here.

Reply:
Good catch, this was a mistake, it should have been testing for `ReaderCreated` only.